tests/test_warp.py

```diff
@@ -1020,41 +1020,54 @@ def test_reproject_crs_none():
             resampling=Resampling.nearest)
 
 
-def test_reproject_identity():
-    """Reproject with an identity matrix."""
-    # note the affines are both positive e, src is identity
+def test_reproject_identity_src():
+    """Reproject with an identity like source matrices."""
     src = np.random.random(25).reshape((1, 5, 5))
-    srcaff = Affine(1.0, 0.0, 0.0, 0.0, 1.0, 0.0)  # Identity
-    srccrs = {'init': 'epsg:3857'}
-
     dst = np.empty(shape=(1, 10, 10))
     dstaff = Affine(0.5, 0.0, 0.0, 0.0, 0.5, 0.0)
-    dstcrs = {'init': 'epsg:3857'}
+    crs = {'init': 'epsg:3857'}
+
+    src_affines = [
+        Affine(1.0, 0.0, 0.0, 0.0, 1.0, 0.0),  # Identity both positive
+        Affine(1.0, 0.0, 0.0, 0.0, -1.0, 0.0),  # Identity with negative e
+        Affine(-1.0, 0.0, 0.0, 0.0, 1.0, 0.0),  # Identity with negative a
```

Comment:
This and the following condition can now be dropped.

Reply:
OK